pages/blog/the-reference-rabbit-hole.md

```diff
@@ -0,0 +1,275 @@
+
+Firstly this is only allowed because we don't first validate the document. However, the underlying behavior still shows that the tooling did not follow the expected behavior, as they include the extra keywords anyway. More precisely, the referenced schema did not overwrite anything, as `type: object` was preserved. For more information see [parser-js #404](https://github.com/asyncapi/parser-js/issues/404).
+
+Then what about JSON Schema, does this define a different behavior? The answer to this question can be found [here](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-01#section-8.3):
+
+> All other properties in a "$ref" object MUST be ignored.
+
+Luckily, they both match the same behavior in terms of extra keywords. Both Reference Object and JSON Schema should ignore extra keywords.
+
+But, what if I use one of the newer JSON Schema versions, what then?
+
+## Upgrading to JSON Schema draft 2020-12
+We started to correlate the findings with the feature request from [Maciej](https://github.com/magicmatatjahu) about updating AsyncAPI Schema Object to point towards [JSON Schema Draft 2020-12](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00).
+
+What would this mean for our little `$ref` keywords?
+
+
+OpenAPI have in its most recent version 3.1, switched its default JSON Schema version to Draft 2020-12, the exact feature request for AsyncAPI. This, however, introduced a huge change to how you bundle references, I dont want to spend much time on this as [Ben](https://twitter.com/relequestual) and [Mike](https://twitter.com/PermittedSoc) described this entire change and what it means in terms of bundling in this great blog post: https://json-schema.org/blog/posts/bundling-json-schema-compound-documents#bundling-simple-external-resources. Besides this the release notes for Draft 2020-12 also offers some guidance which can be found here: https://json-schema.org/draft/2020-12/release-notes.html
+
+Besides having a bunch of new keywords that change the referencing behavior, such as `$dynamicRef`, `$dynamicAnchor`, `$anchor`. One of the key differences is that in [JSON Schema draft 2019-09](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-02), they changed their behavior of references where extra keywords are now allowed adjacent to `$ref`.
```

Comment:
If you start with `Besides`, is because I expect you right after mention the "in addition" part of the sentence.
```suggestion
Besides having a bunch of new keywords that change the referencing behavior, such as `$dynamicRef`, `$dynamicAnchor`, `$anchor`, pne of the key differences is that in [JSON Schema draft 2019-09](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-02), they changed their behavior of references where extra keywords are now allowed adjacent to `$ref`.
```